Kumi.Game/Online/API/DummyAPIConnection.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Kumi.Game.Online.API.Users;
+using Kumi.Game.Online.Server;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+
+namespace Kumi.Game.Online.API;
+
+public partial class DummyAPIConnection : Component, IAPIConnectionProvider
+{
+    public ServerConfiguration EndpointConfiguration { get; } = new DummyServerConfiguration();
+
+    public Queue<APIRequest> RequestQueue { get; private set; } = new Queue<APIRequest>();
+
+    // For testing.
+    internal bool PauseQueue = false;
+
+    public readonly Bindable<APIAccount> LocalAccount = new Bindable<APIAccount>(new GuestAccount());
+    
+    public readonly Bindable<APIState> State = new Bindable<APIState>();
+
+    public string SessionToken { get; } = "dummy";
+
+    /// <summary>
+    /// Passes a function to be executed on the game's update thread, whenever a request is made and ready to be executed.
+    /// </summary>
+    public Func<APIRequest, bool>? HandleRequest { get; set; }
+
+    public new void Schedule(Action action) => base.Schedule(action);
+
+    public DummyAPIConnection()
+    {
+        serverConnector = new ServerConnector(this, false);
+    }
+
+    private IServerConnector? serverConnector;
+
+    public void Queue(APIRequest request)
+    {
+        RequestQueue.Enqueue(request);
+    }
+
+    public void ForceDequeue(APIRequest request)
+    {
+        // reconstruct the queue to place the request at the front.
+        var queue = new Queue<APIRequest>();
+        queue.Enqueue(request);
+        
+        while (RequestQueue.Count > 0)
+            queue.Enqueue(RequestQueue.Dequeue());
+        
+        RequestQueue = queue;
+    }
+
+    public void Perform(APIRequest request)
+        => perform(request);
+
+    public void PerformAsync(APIRequest request)
+        => Task.Factory.StartNew(() => perform(request));
+    
+    public void Login(string username, string password)
+    {
+        State.Value = APIState.Connecting;
+
+        // Create a new account object.
+        LocalAccount.Value = new APIAccount
+        {
+            Id = 1,
+            Username = "Dummy",
+        };
```

Comment:
May want to use `GuestAccount` instead...

Reply:
This is just a fake account upon login, to show that it isn't just the default guest account used only when not logged in.